### PR TITLE
Workaround high memory usage in binary logger

### PR DESCRIPTION
**Pull request by natemcmaster**
Workaround https://github.com/Microsoft/msbuild/issues/2168

cc @KirillOsenkov

### CODE REVIEW

File: build/KoreBuild.ps1 [hunk 1]
```diff
@@ -122,12 +122,14 @@ $msbuildArtifactsDir = "$repoFolder/artifacts/msbuild"
...
 
+# Workaround https://github.com/Microsoft/msbuild/issues/2168
+$sdkPath = ((dotnet --info | sls 'Base Path') -split ':' | select-object -last 1).Trim()
 
 $msBuildArguments = @"
 /nologo
 /m
 /p:RepositoryRoot="$repoFolder/"
-/bl:"$msbuildLogFilePath"
+"/logger:BinaryLogger,$sdkPath/Microsoft.Build.dll:$msbuildLogFilePath"
```
Comment (KirillOsenkov) [inline build/KoreBuild.ps1]: I think that should be a semicolon, not a colon

Comment (KirillOsenkov) [reply build/KoreBuild.ps1]: Also should the full path to the .dll be in quotes? In case there's a space?

**Comment by KirillOsenkov** [comment]
LGTM just fix the colon and quotes. Did you try it locally?

File: build/KoreBuild.ps1 (hunk 1 above)
Comment (natemcmaster) [reply build/KoreBuild.ps1]: Good catch. Windows machine is currently fried so I'm relying on CI + friends to tell me where I've messed up

**Comment by natemcmaster** [comment]
Passes locally on macos. @smitpatel look ok on Windows?

**Comment by smitpatel** [comment]
It will take about half an hour for travis to produce results. I am uncertain about mac but Ubuntu has the worst performance around to an extent causing compile time failure. Let's just wait for EF build to finish.

**Comment by smitpatel** [comment]
EF passed compilation phase. Good to go!

File: build/KoreBuild.ps1 [hunk 2]
```diff
@@ -122,12 +122,14 @@ $msbuildArtifactsDir = "$repoFolder/artifacts/msbuild"
 $msbuildLogFilePath = "$msbuildArtifactsDir/msbuild.binlog"
 $msBuildResponseFile = "$msbuildArtifactsDir/msbuild.rsp"
 
+# Workaround https://github.com/Microsoft/msbuild/issues/2168
+$sdkPath = ((dotnet --info | sls 'Base Path') -split ':' | select-object -last 1).Trim()
```
Comment (smitpatel) [inline build/KoreBuild.ps1:126]: We also have `DotnetInstallationFolder\sdk\DotnetVersion\` to figure out this path. 😉 
Either way is fine though.

**Review by smitpatel — COMMENTED** [review]
Hold on merging. I could not build EF locally using this. Investigating.

**Comment by smitpatel** [comment]
Seems like formatting issue.

```
MSBUILD : error MSB1021: Cannot create an instance of the logger. The given assembly name or codebase was invalid. (Exception from HRESULT: 0x80131047)
Switch: BinaryLogger,\Users\smpatel\.dotnet\x64\sdk\2.0.0-preview2-006215\/Microsoft.Build.dll;D:\EntityFramework/artifacts/msbuild/msbuild.binlog
```

**Comment by KirillOsenkov** [comment]
Yes C: is missing and \/ is weird.

**Comment by smitpatel** [comment]
splitting this `Base Path:   C:\Users\smpatel\.dotnet\x64\sdk\2.0.0-preview2-006215\` wasn't a good idea. Also trailing `\`

**Comment by natemcmaster** [comment]
Discussed with team. We're going to just turn it off for now. 

Closing in favor of https://github.com/aspnet/KoreBuild/pull/256/

**Pull request closed without merging**